tools/distpartitioning/parmetis_postprocess.py

```diff
@@ -39,6 +40,7 @@ def post_process(params):
     )
     global_nids = metis_df["f0"].to_numpy()
     partition_ids = metis_df["f1"].to_numpy()
+    num_parts = np.unique(partition_ids).size
```

Comment:
Is this time consuming?

Reply:
It's about O(n) as the `num_parts` is much smaller than `n`. considering existing operations below, I think it's acceptable.